File: cs1020/miscellaneous/practice-s01p01/Freezer.java
import java.text.DecimalFormat;
import java.util.Scanner;
class Freezer {

    private boolean isTest = false;

    // Computes value of small t as part of formula
    // Precon: User has not entered values of hours and minutes
    // Postcon: Values that user entered has been correctly assigned to hours and minutes
    private double computesSmallT() {
        Scanner sc = new Scanner(System.in);
    
        System.out.print("Enter hours and minutes since power failure: ");
        double hours = sc.nextDouble();
        double minutes = sc.nextDouble(); 
        sc.close();
        
        minutes /= 60; // minutes is assigned value in terms of hours.

        return (hours + minutes);
    }

    // Computes temperature
    // Precon: User has entered values of hours and minutes
    // Postcon: Temperature has been computed and
    private double computesTemperature(double smallT) {
        return ((4.0 * smallT * smallT) / (smallT + 2.0)) - 20.0;
    }

    // Displays temperature
    // Precon: computesTemperature has been computed
    // Postcon: Nil
    private void displaysTemperature(double T) {
        DecimalFormat df = new DecimalFormat("0.00");
        System.out.println("Temperature in freezer = " + df.format(T));
    }

    private void run() {
        double smallT = computesSmallT();
        double T = computesTemperature(smallT);
        displaysTemperature(T);
    }

    public static void main(String[] args) {
        Freezer obj = new Freezer();
        obj.run();
    }
}

File: cs1010/miscellaneous-exercises/practice-s01p01/freezer.c
/*
 *  freezer.c
 *  Unit 4 Exercise 5: Estimate temperature in a freezer
 *  given the elapsed time since power failure.
 *  Formula: T = (4*t^2 / (t+2)) - 20
 */
#include <stdio.h>

int main(void)
{
	int hours, minutes, is_test = 0; // is_test is for debugging
	float temperature;  // Temperature in freezer

	// Get the hours and minutes
	printf("Enter hours and minutes since power failure: ");
	scanf("%d %d", &hours, &minutes);

	float small_t = (float) hours + ((float) minutes / 60);
	if (is_test)
	{
		printf("small_t = %lf\n", small_t);
	}

	temperature = ((4.0 * small_t * small_t) / (small_t + 2.0)) - 20.0;

	printf("Temperature in freezer = %.2f\n", temperature);
	return 0;
}
